apps/frontend/src/app/5_pages/EarnPage/EarnPage.tsx

```diff
@@ -1,12 +1,318 @@
+import { BigNumber } from '@ethersproject/bignumber';
+import {
+  EthersLiquity,
+  ReadableEthersLiquityWithStore,
+} from '@sovryn-zero/lib-ethers';
 
-import { Heading, Paragraph } from '@sovryn/ui';
+import React, { FC, useCallback, useEffect, useMemo, useState } from 'react';
 
-const EarnPage: FC = () => (
-  <>
-    <Heading>Earn</Heading>
-    <Paragraph>This is earn page.</Paragraph>
-  </>
-);
+import classNames from 'classnames';
+import { useTranslation } from 'react-i18next';
+import { useLoaderData } from 'react-router-dom';
+
+import { SupportedTokens } from '@sovryn/contracts';
+import {
+  AmountInput,
+  applyDataAttr,
+  Button,
+  ButtonStyle,
+  ButtonType,
+  Heading,
+  Paragraph,
+  ParagraphSize,
+  Select,
+  SimpleTable,
+  SimpleTableRow,
+  Tabs,
+  TabSize,
+  TabType,
+} from '@sovryn/ui';
+
+import { AssetRenderer } from '../../2_molecules/AssetRenderer/AssetRenderer';
+import { TransactionStepDialog } from '../../3_organisms';
+import { useAccount } from '../../../hooks/useAccount';
+import { useAssetBalance } from '../../../hooks/useAssetBalance';
+import { translations } from '../../../locales/i18n';
+import { formatValue, fromWei, toWei } from '../../../utils/math';
+import { tokenOptions } from './EarnPage.types';
+import { useHandleStabalityDeposit } from './hooks/useHandleStabalityDeposit';
+
+const commonTranslations = translations.common;
+const pageTranslations = translations.earnPage;
+
+const EarnPage: FC = () => {
+  const [index, setIndex] = useState(0);
+  const [amount, setAmount] = useState('0');
+  const [poolBalance, setPoolBalance] = useState('0');
+  const [ZUSDInStabilityPool, setZUSDInStabilityPool] = useState('0');
+  const [token, setToken] = useState<SupportedTokens>(SupportedTokens.dllr);
+  const [isLoading, setIsLoading] = useState(false);
+
+  const { account } = useAccount();
+  const { t } = useTranslation();
+
+  const { liquity } = useLoaderData() as {
+    liquity: EthersLiquity;
+    provider: ReadableEthersLiquityWithStore;
+  };
+
+  const getStabilityDeposit = useCallback(() => {
+    if (account) {
+      setIsLoading(true);
+      liquity
+        .getStabilityDeposit(account)
+        .then(result => setPoolBalance(result.currentZUSD.toString()))
+        .finally(() => setIsLoading(false));
+    }
+  }, [account, liquity]);
+
+  useEffect(() => {
+    getStabilityDeposit();
+  }, [getStabilityDeposit]);
+
+  const getZUSDInStabilityPool = useCallback(() => {
+    liquity
+      .getZUSDInStabilityPool()
+      .then(result => setZUSDInStabilityPool(result.toString()));
+  }, [liquity]);
+
+  useEffect(() => {
+    getZUSDInStabilityPool();
+  }, [getZUSDInStabilityPool]);
+
+  const actions = useMemo(() => {
+    const tabs = [
+      {
+        label: t(commonTranslations.deposit),
+        activeClassName: 'text-primary-20',
+        dataAttribute: 'deposit',
+      },
+    ];
+
+    if (!BigNumber.from(toWei(poolBalance)).isZero()) {
```

Comment:
i believe it's better to move this computation to its own useMemo and pass it as a dependency there